docs/custom-flows/overview.mdx

```diff
@@ -1,5 +1,65 @@
 ---
-sanity_slug: /docs/authentication/custom-flows
+title: Custom flows
+description: Learn the process behind building custom sign-up and sign-in flows with Clerk.
 ---
 
-# This is a stub
+# Custom flows
+
+If Clerk's [prebuilt components](/docs/components/overview) don't meet your specific needs or if you require more control over the authentication flow, Clerk enables you to build fully custom sign-up and sign-in flows using our [`useSignUp()`](/docs/references/react/use-sign-up) and [`useSignIn()`](/docs/references/react/use-sign-in) React hooks.
+
+But before diving deep into the code side, it is important to first understand the process behind building sign-up and sign-in flows, which fields are required, and the necessary steps to authenticate a user.
+
+## Sign-up flow
+
+Clerk provides a flexible way to build sign-up flows in your application. You can use a single [`SignUp`](/docs/references/javascript/sign-up/sign-up) object to gather information, verify their email address or phone number, add OAuth accounts, and finally, convert them into a [`User`](/docs/references/javascript/user/user).
+
+Every [`SignUp`](/docs/references/javascript/sign-up/sign-up) has a set of requirements it must meet before it is turned into a [`User`](/docs/references/javascript/user/user). These requirements are defined by the instance settings you selected in the [Clerk Dashboard](https://dashboard.clerk.com/). Once all requirements are met, the [`SignUp`](/docs/references/javascript/sign-up/sign-up) will turn into a new [`User`](/docs/references/javascript/user/user), and an active session for that [`User`](/docs/references/javascript/user/user) will be created on the current [`Client`](/docs/references/javascript/client).
+
+Don't worry about collecting all the required fields at once and passing them to a single request. The API is designed to accommodate a progressive sign-up flow, often corresponding to multi-step sign-up forms.
+
+### Required fields
+
+The [`SignUp`](/docs/references/javascript/sign-up/sign-up) will show the current state of requirement collection. You can consult the `required_fields`, `optional_fields`, and `missing_fields` keys for a hint on where things are and what you need to do next.
```

Comment:
The first sentence here is a bit confusing to me. What is a "requirement collection"?